src/java.base/share/classes/java/util/concurrent/locks/LockSupport.java

```diff
@@ -457,6 +457,46 @@ static final long getThreadId(Thread thread) {
         return thread.threadId();
     }
 
+    /**
+     *  Preallocated exceptions thrown if acquiring or releasing locks
+     *  when OutOfMemory.
+     */
```

Comment:
I don't see why this should be necessary. IMSE is thrown before any state changes occur and so it is is fine if the IMSE is replaced by OOME.

Even IE should be safe at the points it is thrown.

Also in both cases you want to see a full and proper stacktrace.

Reply:
Consider this another way, any place you have a `throw x` it must be safe to propagate that exception. It doesn't matter if that is actually `x` or an OOME caused by creating `x`.

Reply:
On further consideration, I agree. The very low probability that an IE or IMSE would enable further handling is not worth the extra startup cost of establishing static exceptions. Updated.